tensorflow/core/lib/io/inputbuffer.cc

```diff
@@ -196,5 +197,45 @@ Status InputBuffer::Seek(int64 position) {
   return Status::OK();
 }
 
+Status InputBuffer::Hint(int64 bytes_to_read) {
+  if (bytes_to_read < 0) {
+    return errors::InvalidArgument("Can't read a negative number of bytes: ",
+                                   bytes_to_read);
+  }
+
+  // The internal buffer is too small. Do nothing.
+  if (bytes_to_read > size_) {
+    return Status::OK();
+  }
+
+  const int64 bytes_remain_in_buf = static_cast<int64>(limit_ - pos_);
+
+  // There are enough data in the buffer. Do nothing.
+  if (bytes_to_read <= bytes_remain_in_buf) {
+    return Status::OK();
+  }
+
+  // Additional read from file is necessary. Make some room.
+  memmove(buf_, pos_, bytes_remain_in_buf);
+  pos_ = buf_;
+  limit_ = buf_ + bytes_remain_in_buf;
+  bytes_to_read -= bytes_remain_in_buf;
+
+  // Read the rest bytes from file.
```

Comment:
internal reviewer asked to replace `rest` with `remaining`

Reply:
Fixed.